src/emulator/downloadableEmulators.ts

```diff
@@ -170,17 +170,22 @@ function _getCommand(
   }
 
   const cmdLineArgs = baseCmd.args.slice();
+  const logMetadata = {
+    metadata: {
+      emulator: { name: emulator },
+    },
+  };
   Object.keys(args).forEach((key) => {
     if (baseCmd.optionalArgs.indexOf(key) < 0) {
-      logger.debug(`Ignoring unsupported arg: ${key}`);
+      EmulatorLogger.log("DEBUG", `Ignoring unsupported arg: ${key}`, logMetadata);
```

Comment:
Instead of adding an argument to every statement this seems like what we actually want is to make the logger non-static and pass the metadata to the constructor.

example:
```js
const logger = new EmulatorLogger({ name: emulator });
// ... later ...
logger.log("DEBUG", `Ignoring unsupported arg: ${key}`);
```

Or we could have a map somewhere of Metadata --> Logger, but I definitely think modifying every call site for the logger is not sustainable.

Reply:
I went back and forth on this as well, making a new logger did feel possible but we'd end up making a bunch of instances of single-use loggers or we'd end up having multiple loggers like in Functions where some logs have function data and some don't. Let's chat irl, I can prolly be convinced.